static/app/views/settings/organizationMembers/organizationMemberDetail.tsx

```diff
@@ -45,165 +55,149 @@ const TWO_FACTOR_REQUIRED = t(
+        addErrorMessage(t('Could not send invite'));
+      },
     }
-
-    this.setState({busy: false});
-  };
-
-  handle2faReset = async () => {
-    const {organization, router} = this.props;
-    const {user} = this.state.member!;
-
-    const requests =
-      user?.authenticators?.map(auth =>
-        removeAuthenticator(this.api, user.id, auth.id)
-      ) ?? [];
-
-    try {
-      await Promise.all(requests);
-      router.push(normalizeUrl(`/settings/${organization.slug}/members/`));
+  );
+
+  const {mutate: reset2fa, isPending: isResetting2fa} = useMutation<unknown>({
+    mutationFn: () => {
+      const {user} = member!;
+      const promises =
+        user?.authenticators?.map(auth => removeAuthenticator(api, user.id, auth.id)) ??
+        [];
+      return Promise.all(promises);
+    },
+    onSuccess: () => {
       addSuccessMessage(t('All authenticators have been removed'));
-    } catch (err) {
+      navigate(`/settings/${organization.slug}/members/`);
+    },
+    onError: error => {
       addErrorMessage(t('Error removing authenticators'));
-      Sentry.captureException(err);
-    }
-  };
+      Sentry.captureException(error);
+    },
+  });
 
-  onAddTeam = (teamSlug: string) => {
-    const teamRoles = [...this.state.teamRoles];
-    const i = teamRoles.findIndex(r => r.teamSlug === teamSlug);
+  const onAddTeam = (teamSlug: string) => {
+    const newTeamRoles = [...teamRoles];
+    const i = newTeamRoles.findIndex(r => r.teamSlug === teamSlug);
     if (i !== -1) {
       return;
     }
 
-    teamRoles.push({teamSlug, role: null});
-    this.setState({teamRoles});
+    newTeamRoles.push({teamSlug, role: null});
+    setTeamRoles(newTeamRoles);
   };
 
-  onRemoveTeam = (teamSlug: string) => {
-    const teamRoles = this.state.teamRoles.filter(r => r.teamSlug !== teamSlug);
-    this.setState({teamRoles});
+  const onRemoveTeam = (teamSlug: string) => {
+    const newTeamRoles = teamRoles.filter(r => r.teamSlug !== teamSlug);
+    setTeamRoles(newTeamRoles);
   };
 
-  onChangeOrgRole = orgRole => this.setState({orgRole});
+  const onChangeOrgRole = (newOrgRole: Member['orgRole']) => {
+    setOrgRole(newOrgRole);
+  };
 
-  onChangeTeamRole = (teamSlug: string, role: string) => {
-    if (!this.hasTeamRoles) {
+  const onChangeTeamRole = (teamSlug: string, role: string) => {
+    if (!hasTeamRoles()) {
       return;
     }
 
-    const teamRoles = [...this.state.teamRoles];
-    const i = teamRoles.findIndex(r => r.teamSlug === teamSlug);
+    const newTeamRoles = [...teamRoles];
+    const i = newTeamRoles.findIndex(r => r.teamSlug === teamSlug);
     if (i === -1) {
       return;
     }
 
-    teamRoles[i] = {...teamRoles[i], role};
-    this.setState({teamRoles});
+    newTeamRoles[i] = {...newTeamRoles[i], role};
+    setTeamRoles(newTeamRoles);
   };
 
-  showResetButton = () => {
-    const {organization} = this.props;
-    const {member} = this.state;
+  const showResetButton = () => {
```

Comment:
do we want to use `useCallback` for these functions?

Reply:
reduced a few functions, but i'm not super worried about rerendering here